streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/service/StreamCatalogService.java

```diff
@@ -1088,7 +1107,6 @@ public CustomProcessorInfo removeCustomProcessorInfoAsBundle(String name) throws
             }
         }
         CustomProcessorInfo customProcessorInfo = new CustomProcessorInfo();
-        deleteFileFromStorage(customProcessorInfo.fromTopologyComponentBundle(customProcessorBundle).getJarFileName());
```

Comment:
When are uploaded custom processor jars removed? We may want to remove them when none of the custom processors refer an uploaded jar.

Reply:
Previously jar file name was expected to be unique and provided by user. So i deleted them without any check. Now, as you mentioned it is shared and can be deleted only when its not referenced by any.  That would need an extra query to check. At the same time what is some other connection sends a POST or PUT request for a CP that has a jar with the same digest. Its hard to control that scenario. Need to think through. Moreover, this was modeled after UDFs and Notifiers which also dont delete the jar files if its not referenced anymore. It was discussed in design that we will leave the jar files for now. We can address deleting the jar files later for both UDFs and CPs.

Reply:
Can we file github issue to address this?

Reply:
Raised https://github.com/hortonworks/streamline/issues/1056